### PR TITLE
GH-13: Add steam log in option

### PR DESCRIPTION
**Pull request by tiemonl**
- **achievemint-13: add steam log in**
- **update  callback to use env URL**
- **GH-13: Use session in GET games call**
- **remove local storage config and update kanban call**

### CODE REVIEW

**Comment by lcombs15** [comment]
```tsx
interface SteamSession {
    user?: {
        steam?: {
            steamid?: string
        }
    }
}

export default function Home() {
    const session = useSession()
    const sessionData = session.data as unknown as SteamSession
```

File: src/app/auth.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            jwt({ token, account, profile }) {
+                if (account?.provider === PROVIDER_ID) {
+                    token.steam = profile
+                }
+                return token
+            },
+            session({ session, token }) {
+                if ('steam' in token) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-expect-error
```
Comment (lcombs15) [inline src/app/auth.ts:26]: Do the other types allow us to remove this ignore?

File: src/components/welcome-landing/welcome-landing.tsx [hunk 2]
```diff
@@ -1,61 +1,36 @@
...
+export default function Home() {
+    const session = useSession()
 
-    const router = useRouter();
+    const sessionLayout = <div>
+        <ul>
+            <li>Session status: {session.status}</li>
+            <li>
+                Session data:
+                <pre>{JSON.stringify(session.data, null, 2)}</pre>
```
Comment (lcombs15) [inline src/components/welcome-landing/welcome-landing.tsx:15]: Should we remove this debug info before we merge?

**Review by lcombs15 — APPROVED** [review]

File: src/app/auth.ts (hunk 1 above)
Comment (tiemonl) [reply src/app/auth.ts:26]: Yep, new types allow us to remove it. removed.

File: src/components/welcome-landing/welcome-landing.tsx (hunk 2 above)
Comment (tiemonl) [reply src/components/welcome-landing/welcome-landing.tsx:15]: leaving as is for now. next pr will remove it with a proper log in button and show some sort of info when logged in.

**Pull request merged**